File: misc/benchmarks/programs/node/selection-sort.js
let sort = function(arr) {
    for (let i=0; i < arr.length; i=i+1) {
        let min = i;

        // find minimum value
        for (let j=i+1; j < arr.length; j=j+1) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }

        // swap
        let tmp = arr[i];
        arr[i] = arr[min];
        arr[min] = tmp;
    }
    return arr;
};


const fs = require('fs')
data = fs.readFileSync('numbers.txt', 'utf8')
data = data.split(',').map(d => parseInt(d));

console.log(sort(data));

File: misc/benchmarks/programs/lua/selection-sort.lua
function sort( f )
    for k = 1, #f-1 do   
        -- find idx of min value
        local idx = k    
        for i = k+1, #f do
            if f[i] < f[idx] then 
                idx = i
            end    
        end

        -- swap
        tmp = f[k]
        f[k] = f[idx]
        f[idx] = tmp 
    end
end

-- read entire file 
local file = io.open("numbers.txt", "r") 
local content = file:read "*a"
file:close()

-- split by comma, convert to number
local t={}
for str in string.gmatch(content, "([^, ]+)") do
    table.insert(t, tonumber(str))
end

sort(t)

print(t[1], " - ", t[5000])